SConstruct

```diff
@@ -262,11 +262,11 @@ def addCppLibrary(env, name, dirs=[], tars=[], untarTargets=['configure'], patte
     env.Alias(name, lastTarget)
 
     # Compilation database for libraries and shared libraries
-    env['STATIC_AND_SHARED_OBJECTS_ARE_THE_SAME'] = 1
-    env.SetOption('warn', 'no-duplicate-environment')
-    env.Tool('compilation_db')
+    env2['STATIC_AND_SHARED_OBJECTS_ARE_THE_SAME'] = 1
+    env2.SetOption('warn', 'no-duplicate-environment')
+    env2.Tool('compilation_db')
     xmipp_path = os.path.dirname(os.path.dirname(os.getcwd()))
-    cdb = env.CompilationDatabase(os.path.join(xmipp_path, 'compile_commands.json'))
+    cdb = env2.CompilationDatabase(os.path.join(xmipp_path, 'compile_commands_2.json'))
```

Comment:
2 questions:

- Why does a second env `env2` need to be created?

- Could the `JSON` be renamed to something that does not start with compile, and it is descriptive about it being for xmippCore? This is to avoid clashing with `compileLog.txt` (makes it easier to just type `com` and press tab for autocomplete, which is a bit longer with this file present). Thinking about it, it would also make sense to apply a similar change to the normal xmipp repo.

- Add this file to the `.gitignore`.

Reply:
Hi Martin:

- The second env was already there, I have not created it... I guess it should be this way so Xmipp can compile properly.
- Starting with `compile_` is needed by some IDEs (Fleet, CLion...) as they detect automatically that file name for the indexing. As far as I know, if the name is changed indexing will be broken.
- The file `compile_commands_2.json" is merged after compilation with `compiled_commands.json` once the compilation has finished and removed. Therefore, only `compiled_commands.json` will be kept (already added to `.gitignore`).